cpp/src/io/parquet/reader_impl.cu

```diff
@@ -14,1425 +14,18 @@
  * limitations under the License.
```

Comment:
maybe this one can also be a cpp file now, looks like device code was moved to preprocess.

Reply:
I tried to have all these `cpp` files. Unfurtunately all them indirectly include `parquet_gpu.hpp` which inturn include `cuco/static_map.cuh` that requires to be compiled with nvcc.

I could refactor that header out to make `parquet_gpu.hpp` totally freed from nvcc but that will also touch parquet writer and not sure if everyone will support this idea.

Reply:
IMO `parquet_gpu.hpp` name is somewhat contradict.  The suffix `_gpu` means this is a header for GPU kernels, but `.hpp` extension also indicates the opposite.

Reply:
Darn. Thanks for trying to make this happen.
Why does the helper header need to include `parquet_gpu.cuh`?

> IMO parquet_gpu.hpp name is somewhat contradict. The suffix _gpu means this is a header for GPU kernels, but .hpp extension also indicates the opposite.

I guess that makes sense. The implementations of the functions declared in it do need to be in a `.cu` file.

Reply:
Please see https://github.com/rapidsai/cudf/pull/12046/commits/c35aebe654dd8f7c09596e1af4251e9a833b06e8.

Reply:
this is great!